adr/ADR-3_multi_notary_config.md

```diff
@@ -0,0 +1,103 @@
+
+## Considered options
+
+### Choice 1
+
+The overall notary configuration setup in `helm/values.yaml`.
+
+#### Option 1 (Per Notary)
+
+The `notary` field becomes a list and changes to `notaries`. Per to be used notary instance, there will be one entry in this list.
+
+The entry will have the following data fields (**bold** are mandatory):
+
+- **`name`** -- A unique identifier for the notary configuration, which will be used in the image policy.
+- **`host`** -- The host address of the notary instance.
+- **`root_keys`** -- A list of public root keys, which are to be used for signature verification.
+    - **`name`** -- An identifier for a single public root key, which will be used in the image policy.
+    - **`key`** -- The actual public root key in PEM format.
+- `selfsigned_cert` -- A self-signed certificate in PEM format, for making secure TLS connection to the notary.
+- `auth` -- Authentication details, should the notary require some.
+    - `user` -- Username to authenticate with.
+    - `password` -- Password to authenticate with.
+    - `secretName` -- Kubernetes secret reference to use *INSTEAD* of user/password combination.
+- `is_acr` -- Marks the notary as being part of an Azure Container Registry.
+
+The image policy will have two additional fields per rule entry (in "quotes" are already present fields):
+
+- "`pattern`" -- Image pattern to match against, for rule to apply.
+- "`verify`" -- Whether the images should be verified or not.
+- "`delegations`" -- List of required delegation roles.
+- `notary` -- Which notary to use for any matching image. This has to correspond to a `name` field of one configured notary. What happens if none is given, is defined by the result of choice 2.
+- `key` -- Which key to use for doing the signature verification. This has to correspond to a `name` field of one of the public keys configured for the notary corresponding to the image policy's `notary` field. What happens if none is given, is defined by the result of choice 2.
+
+#### Option 2 (Per Notary + Key)
+
+The `notary` field becomes a list and changes to `notaries`. Per notary + public root key combination, there is one entry. Meaning, for example, there will be one entry for DockerHub and the public key for all official images and there will be another entry for DockerHub and the public key for some private images.
+
+The entries will look identical to the one's from option 1, with two exceptions.
+
+1. The `root_keys` field of the notary configurations won't be a list and only has a single entry, without needing to specify a key name.
+
+2. The image policy will only address the notary configuration to be chosen with the `notary` field, without the need for a `key` field.
+
+### Choice 2
+
+Default values for `notary` (and `key`) inside the image policy.
```

Comment:
I think there might be a 4th option for setting a default:
The image policy was naturally built to allow for setting an overall default via:
```
  - pattern: "*:*"
    verify: true
```
If that specifies the default configuration to use via `notary: default` if not otherwise configured in a more specific rule. If then the preset notary config goes by the name `default` it might be sufficiently explicit:
```
- pattern: "*:*"
  verify: true
  notary: default
- pattern: "k8s.gcr.io/*:*"
  verify: false
- pattern: "docker.io/securesystemsengineering/connaisseur:*"
  verify: false
- pattern: "docker.io/securesystemsengineering/testimage:*"
  verify: true
```
here the last pattern should then just test against the next less specific rule that specifies a notary...